screens/MagicBall.js

```diff
@@ -0,0 +1,158 @@
+import React, { useState, useEffect } from 'react';
+import {
+  View,
+  Text,
+  TouchableOpacity,
+  TextInput,
+  StyleSheet,
+  Alert,
+  ImageBackground,
+} from 'react-native';
+
+import { answers } from './components/answers';
+import { MagicLoader } from '../components/MagicLoader';
+
+const image = {
+  url: 'https://media.istockphoto.com/illustrations/blue-glowing-energy-ball-on-black-background-illustration-id937266238?k=20&m=937266238&s=612x612&w=0&h=JIsUmWm9HzPbZ9LNLeJQpQWknqxCP5jdyalFE6ZNJJU=',
+};
+
+export const MagicBall = ({ navigation }) => {
+  const [magicAnswer, setMagicAnswer] = useState('');
+  const [question, setQuestion] = useState('');
+  const [loading, setLoading] = useState(false);
+  const [newQuestion, setNewQuestion] = useState('');
```

Comment:
Super clearly written useState, I can read very easily what's gonna happen down the road with this 4 lines